From 4dfcc52bce57971b89a7c68596e8b4ce37bf6207 Mon Sep 17 00:00:00 2001
From: Thang Tran <ThangT@fsoft.com.vn>
Date: Wed, 22 Aug 2018 15:18:41 +0700
Subject: [PATCH] HMI:Video_Playback: Remove properties out-width and
 out-height on waylandsink. On VLP2.1, they are replace by property scale on
 waylandsink.

Signed-off-by: Thang Tran <ThangT@fsoft.com.vn>
---
 RZ_GST-demo_video-playback.sh | 34 +++++++++++++++-------------------
 1 file changed, 15 insertions(+), 19 deletions(-)

diff --git a/RZ_GST-demo_video-playback.sh b/RZ_GST-demo_video-playback.sh
index e90e7d7..932f817 100755
--- a/RZ_GST-demo_video-playback.sh
+++ b/RZ_GST-demo_video-playback.sh
@@ -4,17 +4,17 @@ function f_help()
 {
 	printf "\n"
 	printf "\e[32m[Usage]:\e[0m\n"
-	printf "* RZ_GST-demo_video-playback.sh \e[34m<video location> <x> <y> <w> <h>\e[0m\n"
+	printf "* RZ_GST-demo_video-playback.sh \e[34m<video location> <x> <y> <s>\e[0m\n"
 	printf "	- \e[34mvideo location\e[0m: (string) (mandatory) is location of video file.\n"
 	printf "	- \e[34m<x>, <y>\e[0m: (int) (optional) are to set postion of displayed video.\n"
 	printf "	      \e[31mNOTE: <x> and <y> must be set at the same time\e[0m\n"	
-	printf "	- \e[34m<w>, <h>\e[0m: (int) (optional) are to set width and height of video.\n"
-	printf "	      \e[31mNOTE: <w> and <h> must be set at the same time\e[0m\n\n"
+	printf "	- \e[34m<s>\e[0m: (float) (optional) is to set size of video.\n"
+	printf "	      \e[31mNOTE: <s> must be set at the same time\e[0m\n\n"
 	printf "	  \e[31mNOTE: It is posible to use <x>, <y> only\e[0m\n"
 	printf "\n"
 	printf "\e[32m[Example]:\e[0m\n"
-	printf "* Play video at /home/root/videos/video1.mp4 and display at (100, 200) and \nhave the display size is 800x480:\n\n"
-	printf "\e[0mRZ_GST-demo_video-playback.sh \e[34m/home/root/videos/video1.mp4 100 200 800 480\e[0m\n"
+	printf "* Play video at /home/root/videos/video1.mp4 and display at (100, 200) and \nwith scale=0.5 and the display size is 800x480:\n\n"
+	printf "\e[0mRZ_GST-demo_video-playback.sh \e[34m/home/root/videos/video1.mp4 100 200 0.5 \e[0m\n"
 	printf "\n"
 }
 
@@ -28,32 +28,28 @@ if [ "$1" = "--help" ] | [ -z "$1" ]; then
 	f_help
 else
 	unset count_para
-	unset out_width
-	unset out_height
+	unset scale
 	unset pos_x
 	unset pos_y
-	if [ -n "$5" ]; then
-		count_para=4
-		out_width=" out-width=$4"
-		out_height=" out-height=$5"
+	if [ -n "$4" ]; then
+		count_para=3
+		scale=" scale=$4"
 		pos_x=" position-x=$2"
 		pos_y=" position-y=$3"
-	elif [ -n "$4" ]; then
-		count_para=3
 	elif [ -n "$3" ]; then
 		count_para=2
-		pos_x=" position-x=$2"
-                pos_y=" position-y=$3"
 	elif [ -n "$2" ]; then
 		count_para=1
-	else 
-		count_para=0 
+		pos_x=" position-x=$2"
+                pos_y=" position-y=$3"
+	elif [ -n "$1" ]; then
+		count_para=0
 	fi
-	if [ `expr $count_para % 2` -eq 0 ]; then
+	if [ `expr $count_para % 2` -eq 1 ]; then
 		gst-launch-1.0 -e --padprobe probed_pad:sink --timer \
 		filesrc location=$1 ! \
 		qtdemux ! h264parse ! omxh264dec ! \
-		waylandsink $out_width $out_height \
+		waylandsink $scale \
 		$pos_x $pos_y \
 		name=probed_pad qos=false
 	else
-- 
2.7.4

